nvdaHelper/local/UIAEventLimiter/rateLimitedEventHandler.h

```diff
@@ -0,0 +1,85 @@
+/*
+This file is a part of the NVDA project.
+URL: http://github.com/nvaccess/nvda/
+Copyright 2023 NV Access Limited.
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 2.0, as published by
+    the Free Software Foundation.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+This license can be found at:
+http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+*/
+
+#include <vector>
+#include <map>
+#include <list>
+#include <mutex>
+#include <uiautomation.h>
+#include "eventRecord.h"
+
+
+// @brief a class that listens for various UI Automation events, 
+// stores them in in an internal queue (removing any duplicates),
+// and sends them onto an existing UI Automation event handler in a separate thread.
+// This ensures that UI Automation core is never blocked while sending events to this class.
+class RateLimitedEventHandler: public IUIAutomationEventHandler, public IUIAutomationFocusChangedEventHandler, public IUIAutomationPropertyChangedEventHandler, public IUIAutomationNotificationEventHandler, public IUIAutomationActiveTextPositionChangedEventHandler {
```

Comment:
this might be worth splitting into new lines for readability

Reply:
Like this?
```suggestion
class RateLimitedEventHandler:
	public IUIAutomationEventHandler,
	public IUIAutomationFocusChangedEventHandler,
	public IUIAutomationPropertyChangedEventHandler,
	public IUIAutomationNotificationEventHandler,
	public IUIAutomationActiveTextPositionChangedEventHandler
{
```

Reply:
yes, thanks